src/femr/labelers/omop.py

```diff
@@ -463,178 +594,151 @@ def get_labeler_type(self) -> LabelType:
-##########################################################
+        outcome_times = []
+        for idx, row in patient_df.iterrows():
+            label_time = row["start"]
+            label_time = datetime.datetime.strptime(label_time, "%Y-%m-%d %H:%M:%S")
+            prediction_time = label_time - timedelta(hours=24)
 
+            if prediction_time <= start_time:
+                continue
+            outcome_times.append(label_time)
 
-class OpioidOverdoseLabeler(TimeHorizonEventLabeler):
-    """
-    TODO - check
-    The opioid overdose labeler predicts whether or not an opioid overdose will occur in the time horizon
-    after being prescribed opioids.
-    It is conditioned on the patient being prescribed opioids.
-    """
+        return outcome_times
 
-    def __init__(self, ontology: extension_datasets.Ontology, time_horizon: TimeHorizon):
-        self.time_horizon: TimeHorizon = time_horizon
+    def get_prediction_times(self, patient: Patient) -> List[datetime.datetime]:
+        outcome_times = self.get_outcome_times(patient)
+        return [outcome_time - timedelta(hours=24) for outcome_time in outcome_times]
 
-        icd9_codes: List[str] = [
-            "E850.0",
-            "E850.1",
-            "E850.2",
-            "965.00",
-            "965.01",
-            "965.02",
-            "965.09",
-        ]
-        icd10_codes: List[str] = ["T40.0", "T40.1", "T40.2", "T40.3", "T40.4"]
+    def get_labeler_type(self) -> LabelType:
+        return "categorical"
 
-        self.overdose_codes: Set[str] = set()
-        for code in icd9_codes:
-            self.overdose_codes |= _get_all_children(ontology, "ICD9CM/" + code)
-        for code in icd10_codes:
-            self.overdose_codes |= _get_all_children(ontology, "ICD10CM/" + code)
+    def label(self, patient: Patient, patient_df: pd.DataFrame) -> List[Label]:  # type: ignore
+        labels: List[Label] = []
 
-        self.opioid_codes = _get_all_children(ontology, "ATC/N02A")
+        patient_df = patient_df.sort_values(by=["start"], ascending=True)
+        start_time, _ = self.get_patient_start_end_times(patient)
 
-    def get_outcome_times(self, patient: Patient) -> List[datetime.datetime]:
-        """Return the start times of this patient's events whose `code` is in `self.overdose_codes`."""
-        times: List[datetime.datetime] = []
-        for event in patient.events:
-            if event.code in self.overdose_codes:
-                times.append(event.start)
-        return times
-
-    def get_prediction_times(self, patient: Patient) -> List[datetime.datetime]:
-        """Return a sorted list containing the datetimes at which we'll make a prediction."""
-        times: List[datetime.datetime] = []
-        for event in patient.events:
-            if event.code in self.opioid_codes:
-                times.append(event.start)
-        return times
+        for idx, row in patient_df.iterrows():
```

Comment:
Why does this code block need to repeat the get_outcome_times logic?

Reply:
Gonna tag @rthapa84 for the CheXpert related stuff since I don't want to inaccurately characterize anything.